packages/block-editor/src/components/block-toolbar/unified.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { Popover } from '@wordpress/components';
+
+/**
+ * Internal dependencies
+ */
+import { store as blockEditorStore } from '../../store';
+import BlockContextualToolbar from '../block-list/block-contextual-toolbar';
+
+export default function UnifiedBlockToolbar() {
+	const hasFixedToolbar = useSelect(
+		( select ) => select( blockEditorStore ).getSettings().hasFixedToolbar,
+		[]
+	);
+
+	if ( hasFixedToolbar ) {
```

Comment:
Testing this PR and on mobile viewports (with Top Toolbar disabled), there's no block toolbar. 

My understanding is that the desired behavior is for the fixed toolbar to show on mobile viewports.

I think there are still a few files that have some viewport logic for hiding the floating toolbar on mobile viewports (BlockPopover, BlockToolbar).

I wonder if as much of that viewport logic could be consolidated into one place, possibly this component and this `if` statement.

One of the difficult parts of implementing an editor is that the mobile viewport doesn't work out of the box, so it'd be nice if `UnifiedBlockToolbar` helped solve that.

At the same time, if an implementor wants the toolbar to behave differently to the post editor, that's difficult to do as the viewport logic is implicit to much of the block editor package. I think ideally `BlockToolbar` would be just a plain configurable toolbar that we build abstractions around.